### DISCUSSION TITLE
Save resolved execution time parameters in flyteadmin

### DISCUSSION THREAD

**Discussion by katrogan** [flyteorg/flyte#3934] [RFC Incubator]
At create execution time, flyteadmin goes through a series of [fallbacks](https://github.com/flyteorg/flyteadmin/blob/04bcb1539355a7991e1158acd8be000fe246ac2e/pkg/manager/impl/execution_manager.go#L343-L425) to resolve runtime parameters in order of

1. The CreateExecution request spec itself
2. The launch plan spec
3. Workflow matchable attribute overrides
4. Project matchable attribute overrides 
5. The root application defaults

Currently, we only save the [CreateExecution request spec](https://github.com/flyteorg/flyteadmin/blob/04bcb1539355a7991e1158acd8be000fe246ac2e/pkg/manager/impl/execution_manager.go#L894) parameters in the db. We should save the final resolved value of the merged in workflow execution config in a separate blob in the db. It's useful to capture the request spec params themselves but it's hard to reason about what the actual runtime values are after the fact.

Ideally we save at what layer in all the override logic all the parameters themselves were sourced from for traceability and debuggability

**Comment by davidmirror-ops** [comment]
2023-11-09 Contributor's meetup notes: this entry is a better fit for an Issue, instead of an RFC. @katrogan feel free to create Issue from this discussion. Thanks!
